tests/decorators/tests.py

```diff
@@ -90,6 +103,64 @@ def test_attributes(self):
         self.assertEqual(fully_decorated.__doc__, 'Expected __doc__')
         self.assertEqual(fully_decorated.__dict__['anything'], 'Expected __dict__')
 
+    def test_escaping(self):
+        """
+        Tests that safety markers from django.utils.safestring
+        return the proper str or unicode subclass for use in templates.
+
+        Tests are done by comparing directly a sample rendered Template instance
+        with the unicode content it should contain.
+        """
+        rendered = {'safe': '<html><body>dummy</body></html>',
+                    'escaped': '&lt;html&gt;&lt;body&gt;dummy&lt;/body&gt;&lt;/html&gt;'}
+
+        def clean_string_provider():
+            return '<html><body>dummy</body></html>'
+
+        with ignore_warnings(category=RemovedInDjango20Warning):
+            escaped_str = mark_for_escaping(clean_string_provider)()
+
+        safe_str = mark_safe(clean_string_provider)()
+
+        self.assertRenderEqual('{{ s }}', rendered['escaped'], s=escaped_str)
+        self.assertRenderEqual('{{ s }}', rendered['safe'], s=safe_str)
+
+    def test_mark_safe_decorator_does_not_affect_dunder_html(self):
+        """
+        Test that after implementing the mark_safe decorator, we don't break previous behavior of honoring objects
```

Comment:
Wrap docstrings at 79 characters. Try to avoid "we... " (often this results in simpler language).